File: Leetcode/0938 Range Sum of BST.py
class Solution:
    def rangeSumBST(self, root, low, high):
        if root is None or low > high:
            return 0

        range_sum = root.val if low <= root.val <= high else 0

        if not (root.val < low):
            range_sum += self.rangeSumBST(root.left, low, min(root.val, high))
        if not (root.val > high):
            range_sum += self.rangeSumBST(root.right, max(root.val, low), high)

        return range_sum

File: Leetcode/0938 Range Sum of BST.ts
function rangeSumBST(root: TreeNode | null, low: number, high: number): number {
    if (root === null || low > high) {
        return 0;
    }
    let rangeSum = low <= root.val && root.val <= high? root.val : 0;
    if (root.val >= low) {
        rangeSum += rangeSumBST(root.left, low, Math.min(high, root.val));
    }
    if (root.val <= high) {
        rangeSum += rangeSumBST(root.right, Math.max(root.val, low), high);
    }
    return rangeSum;
}
